receiver/podmanreceiver/receiver.go

```diff
@@ -57,11 +67,84 @@ func newReceiver(
 		set:           set,
 	}
 
-	scrp, err := scraperhelper.NewScraper(typeStr, recv.scrape, scraperhelper.WithStart(recv.start))
+	return recv, err
+}
+
+func (r *receiver) registerMetricsConsumer(mc consumer.Metrics, set component.ReceiverCreateSettings) error {
+	r.metricsConsumer = mc
+	scrp, err := scraperhelper.NewScraper(typeStr, r.scrape, scraperhelper.WithStart(r.start))
 	if err != nil {
-		return nil, err
+		return err
+	}
+	r.metricsComponent, err = scraperhelper.NewScraperControllerReceiver(&r.config.ScraperControllerSettings, set, mc, scraperhelper.AddScraper(scrp))
+	return err
+}
+
+func (r *receiver) registerLogsConsumer(lc consumer.Logs) {
+	r.logsConsumer = lc
+}
+
+func (r *receiver) Start(ctx context.Context, host component.Host) error {
```

Comment:
Sorry for not catching this earlier, but you have two functions with the same name, one public and one exported:

```
func (r *receiver) Start(ctx context.Context, host component.Host) error {
func (r *receiver) start(context.Context, component.Host) error {
```

Considering for a moment that the exported one should indeed be exported, one of them should still be renamed, as it's not clear from a first look what's the relationship between them, as one doesn't (directly?) call the other.

Reply:
We have observed in this repo that `start` is used as a method name where `scraperhelper` is used. Also, we need `Start` method to start the receiver. So, to avoid confusion, we can add an informative comment on both methods.

Reply:
Since you introduced `Start` explicitly and moved receiver created by scraperhelper into `metricsComponent`, you don't need the `start` callback, you can move the code from `start` to `Start`

Reply:
moved the code of `start` to `Start` and removed `start` method as it is no longer needed.